### PR TITLE
Modern theme styles for summary action bar component

### PR DESCRIPTION
**Pull request by Blackbaud-AlexKingman**
Notes:

1. Dropdown button types don't support skyux fonts yet. We've filed an issue for that. Should not be a blocker.
~2. Key info component needs updated padding for modern theme. Not a blocker, but that work is done [here](https://github.com/blackbaud/skyux-indicators/pull/163).~ (completed)

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-action-bars?_sp.spa=pr-modern-theme](https://developer.blackbaud.com/skyux-action-bars?_sp.spa=pr-modern-theme)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-action-bars?_sp.spa=pr-modern-theme](https://developer.blackbaud.com/skyux-action-bars?_sp.spa=pr-modern-theme)

**Comment by Blackbaud-AlexKingman** [comment]
@Blackbaud-ToddRoberts @Blackbaud-AdamFunderburk When there's a lot of action buttons, it can force the expand/collapse button to run into the bottom of the container. This has always existed (you can force it in default), but didn't really show until modern theme, b/c there is naturally more spacing in the buttons. Is this a problem we should address?

Screen Shot 2021-04-13 at 9 58 53 AM

**Comment by Blackbaud-ToddRoberts** [comment]
I thought secondary buttons collapsed into a context menu if there wasn't enough space, to avoid this?

**Comment by Blackbaud-AlexKingman** [comment]
> I thought secondary buttons collapsed into a context menu if there wasn't enough space, to avoid this?

Looked into it... Seems like that functionality only works when the screen width is under `767px`. It does not work for modals until we add responsive styles to the modal :-/

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-action-bars?_sp.spa=pr-modern-theme](https://developer.blackbaud.com/skyux-action-bars?_sp.spa=pr-modern-theme)

**Comment by Blackbaud-TrevorBurch** [comment]
> > I thought secondary buttons collapsed into a context menu if there wasn't enough space, to avoid this?
> 
> Looked into it... Seems like that functionality only works when the screen width is under `767px`. It does not work for modals until we add responsive styles to the modal :-/

@Blackbaud-AlexKingman - I thought that we already added those to modals?

**Comment by Blackbaud-TrevorBurch** [comment]
> > I thought secondary buttons collapsed into a context menu if there wasn't enough space, to avoid this?
> 
> Looked into it... Seems like that functionality only works when the screen width is under `767px`. It does not work for modals until we add responsive styles to the modal :-/

@Blackbaud-AlexKingman - I thought that was already done but I don't see anything in the modals code for it. Could you file a bug for that so that we can triage it? This can't be the only thing being affected negatively but not having those styles.

**Comment by Blackbaud-TrevorBurch** [comment]
@Blackbaud-AlexKingman - it appears we are resetting old visual screenshots for default - why are we doing that?

File: src/app/visual/summary-action-bar/summary-action-bar-demo.component.html [hunk 1]
```diff
@@ -81,5 +85,61 @@
         <sky-key-info-label>Field 3</sky-key-info-label>
       </sky-key-info>
     </div>
+    <div *skyThemeIf="'modern'">
```
Comment (Blackbaud-TrevorBurch) [inline src/app/visual/summary-action-bar/summary-action-bar-demo.component.html:88]: Is the only difference here the `layout`? I know that we use the `skyThemeIf` for most things - but would it just be better to grab the theme and conditional the `layout` on these? It seems like a lot of extra repeated code just for `layout`.

**Comment by Blackbaud-AlexKingman** [comment]
> @Blackbaud-AlexKingman - it appears we are resetting old visual screenshots for default - why are we doing that?

The change we're making to key info spacing will cause them to fail. I just got ahead of it. As soon as it releases, I'll pull it in here.

**Review by Blackbaud-TrevorBurch — CHANGES_REQUESTED** [review]
@Blackbaud-AlexKingman - just a few questions for you

**Review by Blackbaud-ToddRoberts — CHANGES_REQUESTED** [review]
key info should use xl horizontal spacing (30px) - not sure if this is built into the component, or our demo.

File: e2e/summary-action-bar-in-split-view.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  function runTests(): void {
+    it('should match previous screenshot', async (done) => {
+      await SkyHostBrowser.scrollTo('#screenshot-split-view');
+      expect('#screenshot-split-view').toMatchBaselineScreenshot(done, {
+        screenshotName: getScreenshotName('summary-action-bar-split-view')
+      });
+    });
+  }
+
+  describe('(size: lg)', () => {
```
[the remark this answers is not in the document]
Comment (Blackbaud-AlexKingman) [reply e2e/summary-action-bar-in-split-view.e2e-spec.ts:55]: Added.

File: e2e/summary-action-bar-in-tabset.e2e-spec.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+    it('should match previous screenshot with a summary action bar', async (done) => {
+      await element(by.css('#sky-tab-1-nav-btn')).click();
+      expect('#screenshot-tabset').toMatchBaselineScreenshot(done, {
+        screenshotName: getScreenshotName('summary-action-bar-tab')
+      });
+    });
+  }
+
+  describe('(size: lg)', () => {
```
[the remark this answers is not in the document]
Comment (Blackbaud-AlexKingman) [reply e2e/summary-action-bar-in-tabset.e2e-spec.ts:66]: Added.

File: src/app/visual/summary-action-bar/summary-action-bar-demo.component.html (hunk 1 above)
Comment (Blackbaud-AlexKingman) [reply src/app/visual/summary-action-bar/summary-action-bar-demo.component.html:88]: Yep. Updated.

File: src/app/public/modules/summary-action-bar/summary-action-bar.component.html [hunk 4]
```diff
@@ -33,16 +33,26 @@
...
             [attr.aria-controls]="summaryId"
             [attr.aria-expanded]="true"
+            [skyThemeClass]="{
+              'sky-btn-secondary' : 'default',
+              'sky-btn-default sky-btn-icon': 'modern'
+            }"
             (click)="hideSummarySection()"
           >
-            <sky-icon
+            <sky-icon *skyThemeIf="'default'"
```
[the remark this answers is not in the document]
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/summary-action-bar/summary-action-bar.component.html:47]: Updated to use the Chevron.

File: src/app/public/modules/summary-action-bar/summary-action-bar.component.html [hunk 5]
```diff
@@ -32,36 +32,34 @@
...
         >
-          <button
-            class="sky-btn sky-btn-secondary"
-            type="button"
+          <sky-chevron
             [attr.title]="'skyux_summary_action_bar_hide_summary' | skyLibResources"
             [attr.aria-controls]="summaryId"
             [attr.aria-expanded]="true"
-            (click)="hideSummarySection()"
+            [direction]="isSummaryCollapsed ? 'up' : 'down'"
```
Comment (Blackbaud-TrevorBurch) [inline src/app/public/modules/summary-action-bar/summary-action-bar.component.html:39]: I don't think we need both `sky-chevron` components now do we? Can't we just use one with the `direction` property? Or am I forgetting some nuance here?

**Review by Blackbaud-TrevorBurch — CHANGES_REQUESTED** [review]
One final question (along with Todd's feedback)

File: src/app/public/modules/summary-action-bar/summary-action-bar.component.html (hunk 5 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/summary-action-bar/summary-action-bar.component.html]: The differing nesting structure of the template based on what mode you're in makes that tricky. To clean things up I put the chevron inside an `ng-template` is that better?

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-action-bars?_sp.spa=pr-modern-theme](https://developer.blackbaud.com/skyux-action-bars?_sp.spa=pr-modern-theme)

**Review by Blackbaud-TrevorBurch — APPROVED** [review]

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-action-bars?_sp.spa=pr-modern-theme](https://developer.blackbaud.com/skyux-action-bars?_sp.spa=pr-modern-theme)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-action-bars?_sp.spa=pr-modern-theme](https://developer.blackbaud.com/skyux-action-bars?_sp.spa=pr-modern-theme)

**Review by Blackbaud-ToddRoberts — APPROVED** [review]

**Review by Blackbaud-AdamFunderburk — APPROVED** [review]

**Review by Blackbaud-TrevorBurch — APPROVED** [review]

**Pull request merged**